scripts/timelapse-colorizer-data/convert_emt_migration_data.py

```diff
@@ -0,0 +1,215 @@
+from aicsimageio import AICSImage
+import argparse
+import json
+import logging
+import numpy as np
+import pandas as pd
+import time
+
+from data_writer_utils import (
+    INITIAL_INDEX_COLUMN,
+    ColorizerDatasetWriter,
+    configureLogging,
+    scale_image,
+    remap_segmented_image,
+)
+
+# DATASET SPEC: See DATA_FORMAT.md for more details on the dataset format!
+# You can find the most updated version on GitHub here:
+# https://github.com/allen-cell-animated/nucmorph-colorizer/blob/main/documentation/DATA_FORMAT.md
+
+# OVERWRITE THESE!! These values should change based on your dataset. These are
+# relabeled as constants here for clarity/intent of the column name.
+OBJECT_ID_COLUMN = "R0Nuclei_Number_Object_Number"
+"""Column of object IDs (or unique row number)."""
+TRACK_ID_COLUMN = "R0Nuclei_TrackObjects_Label_75"
+"""Column of track ID for each object."""
+TIMES_COLUMN = "Image_Metadata_Timepoint"
+"""Column of frame number that the object ID appears in."""
+SEGMENTED_IMAGE_COLUMN = "OutputMask (CAAX)"
+"""Column of path to the segmented image data or z stack for the frame."""
+CENTROIDS_X_COLUMN = "R0Nuclei_AreaShape_Center_X"
+"""Column of X centroid coordinates, in pixels of original image data."""
+CENTROIDS_Y_COLUMN = "R0Nuclei_AreaShape_Center_Y"
+"""Column of Y centroid coordinates, in pixels of original image data."""
+
+
+def make_frames(grouped_frames, scale: float, writer: ColorizerDatasetWriter):
+    """
+    Generate the images and bounding boxes for each time step in the dataset.
+    """
+    nframes = len(grouped_frames)
+    logging.info("Making {} frames...".format(nframes))
+
+    for group_name, frame in grouped_frames:
+        start_time = time.time()
+
+        # Get the path to the segmented zstack image frame from the first row (should be the same for
+        # all rows in this group, since they are all on the same frame).
+        row = frame.iloc[0]
+        frame_number = row[TIMES_COLUMN]
+        # Flatten the z-stack to a 2D image.
+        zstackpath = row[SEGMENTED_IMAGE_COLUMN]
+        zstackpath = zstackpath.strip('"')
+        zstack = AICSImage(zstackpath).get_image_data("YX", S=0, T=0, C=0)
+        seg2d = zstack
+
+        # Scale the image and format as integers.
+        seg2d = scale_image(seg2d, scale)
+        seg2d = seg2d.astype(np.uint32)
+
+        # Remap the frame image so the IDs are unique across the whole dataset.
+        seg_remapped, lut = remap_segmented_image(
+            seg2d,
+            frame,
+            OBJECT_ID_COLUMN,
+        )
+
+        writer.update_and_write_bbox_data(grouped_frames, seg_remapped, lut)
+        writer.write_image(seg_remapped, frame_number)
```

Comment:
is there a reason to have to call these successively? could you make one function in writer that that calls these both?